test/sol6/multipleEpochRewardsClaimer.js

```diff
@@ -0,0 +1,227 @@
+        let initialEpoch;
+        let currentEpoch;
+        let numEpochs = 3;
+
+        before("setup dao, feehandler and rewardsClaimer", async() => {
+            // setup mockKyberDao
+            epoch = new BN(0);
+            expiryTimestamp = new BN(5);
+            mockKyberDao = await MockKyberDao.new(
+                rewardInBPS,
+                rebateInBPS,
+                epoch,
+                expiryTimestamp
+            );
+    
+            stakerPercentageInPrecision = precisionUnits.mul(rewardInBPS).div(BPS);
+            await mockKyberDao.setStakerPercentageInPrecision(stakerPercentageInPrecision);
+    
+            // setup feeHandler
+            feeHandler = await FeeHandler.new(daoSetter, proxy.address, kyberNetwork, knc.address, BURN_BLOCK_INTERVAL, daoOperator);
+            await feeHandler.setDaoContract(mockKyberDao.address, {from: daoSetter})
+            await feeHandler.setBurnConfigParams(sanityRate.address, ethWeiToBurn, {from: daoOperator});
+            await feeHandler.getBRR();
+    
+            // setup rewardsClaimer
+            rewardsClaimer = await MultipleEpochRewardsClaimer.new(mockKyberDao.address);
+        });
+
+        beforeEach("send fees for numEpochs", async() => {
+            initialEpoch = (await mockKyberDao.epoch()).toNumber();
+            await sendFeesToFeeHandler(mockKyberDao, feeHandler, numEpochs);
+            currentEpoch = (await mockKyberDao.epoch()).toNumber();
+        });
+
+        it("should not do anything if empty array is passed", async() => {
+            let stakerBalance = await Helper.getBalancePromise(staker);
+            await rewardsClaimer.claimMultipleRewards(feeHandler.address, [], {from: staker});
+            Helper.assertLesser(await Helper.getBalancePromise(staker), stakerBalance, "staker balance changed");
+        });
+
+        it("should claim for 1 epoch", async() => {
+            let stakerBalance = await Helper.getBalancePromise(staker);
+            await rewardsClaimer.claimMultipleRewards(feeHandler.address, [initialEpoch], {from: staker});
+            Helper.assertGreater(await Helper.getBalancePromise(staker), stakerBalance, "staker balance did not increase");
+            assert.isTrue(await feeHandler.hasClaimedReward(staker, initialEpoch));
+        });
+
+        it("should claim for multiple epochs", async() => {
+            let stakerBalance = await Helper.getBalancePromise(staker);
+            await rewardsClaimer.claimMultipleRewards(feeHandler.address, [initialEpoch, currentEpoch - 2, currentEpoch - 1], {from: staker});
+            Helper.assertGreater(await Helper.getBalancePromise(staker), stakerBalance, "staker balance did not increase");
+            assert.isTrue(await feeHandler.hasClaimedReward(staker, initialEpoch));
+            assert.isTrue(await feeHandler.hasClaimedReward(staker, currentEpoch - 2));
+            assert.isTrue(await feeHandler.hasClaimedReward(staker, currentEpoch - 1));
+        });
```

Comment:
please add one test for 12 epochs and print gas

or is this mock contracts and gas not reliable?

Reply:
to have gas consumption report, we will need to use the real KyberStaking and KyberDao codes

Reply:
mock dao is used... Anyway, we also discussed about gas consumption already, that each claim tx is 87796 gas on mainnet, so we should expect roughly 12 * (87796 - 21000) + 21000 = 822552 (~822k gas), with some additional gas for computation.

Reply:
OK